tests/dataloader_tests.cpp

```diff
@@ -0,0 +1,68 @@
+/**
+ * @file dataloader_tests.cpp
+ * @author Kartik Dutt
+ *
+ * Tests for various functionalities of dataloader.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#define BOOST_TEST_DYN_LINK
+#define BOOST_TEST_MAIN // Do not define this anywhere else.
+#include <dataloader/dataloader.hpp>
+#include <boost/test/unit_test.hpp>
+
+using namespace boost::unit_test;
+
+BOOST_AUTO_TEST_SUITE(DataLoadersTest);
+
+/**
+ * Simple test for Dataloader.
+ */
+BOOST_AUTO_TEST_CASE(CSVDataLoaderTest)
+{
+  // Check that dataloader loads only known datasets.
+  BOOST_REQUIRE_THROW(DataLoader<>("no-dataset", true), std::runtime_error);
+
+  // Check Load CSV Function for Dataloader.
+  Utils::DownloadFile("/datasets/iris.csv", "./../data/iris.csv");
```

Comment:
Can you check if that works with an absolute path? Or I guess `iris.csv` should work as well, maybe the path isn't correct? Looking at the ci config:

`script: cd build/tests/ && sudo CTEST_OUTPUT_ON_FAILURE=1 ctest -R UtilsTest`

we should be in `build/tests/` so I would think we have to use `"./../../data/iris.csv"`?

Reply:
> we should be in build/tests/ so I would think we have to use "./../../data/iris.csv"?

I think this shouldn't be required since CtestConfig.CMake sets path for execution to build so the data folder should be accessible. I'll try with absolute path and get back to you. Thanks a lot.

Reply:
Finally figured it out, The tests pass now (getting 400 in one of them) but the rest pass. The issue was GitHub deleted the empty data folder. Locally I still had the folder so that caused this issue on azure and not locally. Sorry about this.
Thanks a lot.
Regards.

Reply:
Ah, great that you figured it out.

Reply:
I made a lot of commits for this. Hopefully this commit will give a green build on OSX. (Squashing the rest).

Reply:
Right, it's often not that easy to debug something on the ci pipeline.